src/fragmentarium/ui/info/Genre.tsx

```diff
@@ -0,0 +1,165 @@
+import { Fragment } from 'fragmentarium/domain/fragment'
+import React, { useEffect, useRef, useState } from 'react'
+import {
+  Button,
+  Overlay,
+  OverlayTrigger,
+  Popover,
+  Tooltip,
+} from 'react-bootstrap'
+import Select from 'react-select'
+import classNames from 'classnames'
+import { usePrevious } from 'common/usePrevious'
+import withData from 'http/withData'
+import _ from 'lodash'
+import FragmentService from 'fragmentarium/application/FragmentService'
+
+type Props = {
+  fragment: Fragment
+  updateGenre: (genre: any) => void
+  genreOptions: readonly string[][]
+}
+
+function Genre({ fragment, updateGenre, genreOptions }: Props): JSX.Element {
+  const [genre, setGenre] = useState(fragment.genre)
+  const prevGenre = usePrevious(genre)
+  const [isSelectDisplayed, setIsSelectDisplayed] = useState(false)
+  const [
+    isDuplicateWarningDisplayed,
+    setIsDuplicateWarningDisplayed,
+  ] = useState(false)
+  const prevIsDuplicateWarningDisplayed = usePrevious(
+    isDuplicateWarningDisplayed
+  )
+  const target = useRef(null)
+
+  function handleChange(event) {
+    if (!isGenreAlreadyPresent(event.value)) {
+      setGenre((currentGenre) =>
+        _.sortBy([...currentGenre, event.value], function (item) {
+          return JSON.stringify(genreOptions).indexOf(JSON.stringify(item))
+        })
+      )
+      setIsSelectDisplayed(false)
+    } else {
+      setIsDuplicateWarningDisplayed(true)
+    }
+  }
+  useEffect(() => {
+    if (JSON.stringify(prevGenre) !== JSON.stringify(genre)) {
+      updateGenre(genre)
+    }
+  })
+  useEffect(() => {
+    if (!prevIsDuplicateWarningDisplayed && isDuplicateWarningDisplayed) {
+      setTimeout(() => {
+        setIsDuplicateWarningDisplayed(false)
+      }, 2500)
+    }
+  })
```

Comment:
Why time out and what's the thing with the previous warning?

Reply:
the previous warning was indeed unnecessary. If the `isDuplicateWarning` is true, the user sees a popup saying `This option is already selected` which is shown for a short period (thats why we need the timeout).